### ISSUE TITLE
GIProbe compression issues

### ISSUE THREAD

**Issue by ghost** [godotengine/godot#8705]
- [ ] When GIProbe compression is enabled, [this line](https://github.com/godotengine/godot/blob/f527a6946285502f8d518f054b1bebc1d3fa564e/drivers/gles3/rasterizer_storage_gles3.cpp#L4783), which is

```cpp
glCompressedTexImage3D(GL_TEXTURE_3D, level, _EXT_COMPRESSED_RGBA_S3TC_DXT5_EXT, p_width, p_height, p_depth, 0, size, NULL);
```

results in error:

> ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 84  Severity: High  Message: GL_INVALID_ENUM in glCompressedTexImage3D(target)
   At: drivers/gles3/rasterizer_gles3.cpp:123.

Similarly, [this line](https://github.com/godotengine/godot/blob/f527a6946285502f8d518f054b1bebc1d3fa564e/drivers/gles3/rasterizer_storage_gles3.cpp#L4835)

```cpp
glCompressedTexSubImage3D(GL_TEXTURE_3D, p_mipmap, 0, 0, p_depth_slice, gipd->width >> p_mipmap, gipd->height >> p_mipmap, p_slice_count, _EXT_COMPRESSED_RGBA_S3TC_DXT5_EXT, size, p_data);
```

yields

> ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 84  Severity: High  Message: GL_INVALID_OPERATION in glCompressedTexSubImage3D(invalid target GL_TEXTURE_3D for format GL_COMPRESSED_RGBA_S3TC_DXT5_EXT)
   At: drivers/gles3/rasterizer_gles3.cpp:123.

- [ ] It uses an inline DXT5 compressor (see [here](https://github.com/godotengine/godot/blob/master/servers/visual/visual_server_scene.cpp#L2370) and [here](
https://github.com/godotengine/godot/blob/master/servers/visual/visual_server_scene.cpp#L2905)), rather than using libsquish

- [ ] This assumes S3TC support. For mobile, ETC2 could be used.

**Comment by akien-mga** [comment]
Sorry for the late triage; I think part or all of those issues are fixed by now (partly by you), no?

**Comment by tagcup** [comment]
Probably not?

**Comment by reduz** [comment]
no, it's kind of broken and not much hope to fix, will probably redesign
this better for 3.1

**Comment by reduz** [comment]
Kicking to 3.2, I think this compression will eventually be removed because it just does not look good enough.  An alternative could be to use BPTC instead, but will eventually re-evaluate it.

**Comment by akien-mga** [comment]
Fixed by #41471.
